module-2/backend/week_two.md

```diff
@@ -4,35 +4,112 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
+5. What do they allow you to do?<br>
+
+I'm not sure its appropriate to invoke the word "scope", but in a way they allow you to scope down into collections of things based on their relationships. I've been thinking about it like searching from the perspective of the end point of one of the relationships. 
+
+6. What's the difference between agile workflow and waterfall method?<br>
+
+Agile takes as its philosophy that a client never knows what they want until they can see it and that a project should be approached doing a little bit of everything(in terms of design, planning, build-out, testing) developing iteratively. Waterfall is the approach that some enigneers sit down and try to think of how a thing should be at the end down to the last detail. They design it and that it, thats the final design. They then pass it off to another team and the progression goes in strict order.  
+
+7. What is the difference between `#new` and `#create`?<br>
+
+`#new` gives us a blank object with all the attributes being nil. We can use it to check what attributes it has or apparently we can use it as a placeholder for some crazy rails thing called reflection. If we assign the new object to a variable we can fill its attributes and `.save` to make those changes in the database. `#create` does all that at once, makes the object and saves it. 
+
 8. At a basic level, what does cURL allow you to do?
+
+cURL is a command line utitlity that allow you to go to a resource online, grab it and bring it back.
+
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
-10. Define foreign key, primary key, and schema.
-11. Describe the relationship between a foreign key on one table and a primary key on another table.
-12. What are the parts of an HTTP response?
-13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
-14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+
+At its simplest a teacher will have many students and a student belongs to a teacher. In reality perhaps it would be the case of many to many as students have multiple teachers. 
+
+In the simple case the student table would have a foreign key that corresponds to a teacher_id. In a slightly more complex case the teachers and students would be tied through a joins table that would contain id, teacher_id, and student_id. More complex still would be to have the third table have its own attributes, perhaps we could call it curriculum. A teacher would have many students through curriculum and a student would have many teachers through curriculum (curriculum in the sense of program or track). A curriculum would have many students and would have many teachers. 
```

Comment:
Like that you took the time to consider some different options based on how the relationship works.